server/cache.go

```diff
@@ -158,14 +158,17 @@ type regionsInfo struct {
 	regions map[uint64]*metapb.Region
 	// search key -> region id
 	searchRegions *btree.BTree
+	// store id -> region count
+	storeRegionCount map[uint64]uint64
```

Comment:
Because only `regionsInfo` knows how to maintain the `storeRegionCount` properly, and `regionsInfo` already has a `leaderRegionCount` method.